File: backend/app/schemas/summary.py
# app/schemas/summary.py
from typing import Dict, List, Optional
from pydantic import BaseModel


class SegmentMealItem(BaseModel):
    id: str
    meal_text: str
    calories: float


class DaySummary(BaseModel):
    date: str
    day: int
    calories: float
    protein: float
    carbs: float
    fibre: float
    fats: float
    steps: int
    segments: Dict[str, List[SegmentMealItem]]


class MonthlySummaryResponse(BaseModel):
    year: int
    month: int
    days: List[DaySummary]

File: frontend/src/models/summary.ts
// src/models/summary.ts
export interface SegmentMealItem {
  id: string;
  meal_text: string;
  calories: number;
}

export interface DaySummary {
  date: string;
  day: number;
  calories: number;
  protein: number;
  carbs: number;
  fibre: number;
  fats: number;
  steps: number;
  segments: {
    Breakfast: SegmentMealItem[];
    Lunch: SegmentMealItem[];
    Dinner: SegmentMealItem[];
    Snacks: SegmentMealItem[];
  };
}

export interface MonthlySummaryResponse {
  year: number;
  month: number;
  days: DaySummary[];
}
